src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -71,15 +71,24 @@ idx_t IcebergMultiFileList::GetTotalFileCount() {
 }
 
 unique_ptr<NodeStatistics> IcebergMultiFileList::GetCardinality(ClientContext &context) {
-	auto total_file_count = IcebergMultiFileList::GetTotalFileCount();
+	idx_t cardinality = 0;
 
-	if (total_file_count == 0) {
-		return make_uniq<NodeStatistics>(0, 0);
+	if (snapshot.iceberg_format_version == 1) {
+		//! We collect no cardinality information from manifests for V1 tables.
+		return nullptr;
 	}
 
-	// FIXME: visit metadata to get a cardinality count
+	//! Make sure we have fetched all manifests
+	(void)GetTotalFileCount();
```

Comment:
We should look into using the `summary` field of the snapshot (https://iceberg.apache.org/spec/?h=optional+sn#optional-snapshot-summary-fields), probably the `total-records` is the current cardinality

Reply:
the summary fields `total-records` is not entirely accurate, see https://github.com/apache/iceberg/issues/12823

Reply:
Hmm, can a different part of the summary be used for the cardinality estimation?

We're mostly concerned with the cost of expanding the manifests to get this information

Though admittedly this could be optimized to only read the manifest_list, without traversing into the manifests's avro files

Reply:
if its just an estimate, you might get away with using the summary. 

> We're mostly concerned with the cost of expanding the manifests to get this information

`record_count` in the manifest is the correct cardinality. If `GetCardinality` is on the read path, I think you'd have to read the manifest files at a later time anyways. So maybe we can cache it and amortize the cost

Reply:
Thanks, we'll stick with `existing_rows_count` + `added_rows_count` - `deleted_rows_count` 👍

Reply:
This can still be off, since with merge-on-read delete files, rows might be deleted twice since Iceberg uses optimistic concurrency.

Reply:
It's not a problem, it's an estimate